File: src/map.cpp
#include <Eigen/Dense>
#include <vector>
#include "map.hpp"

namespace Map{
Wall::Wall(const Eigen::Vector2d start,
           const Eigen::Vector2d end)
    : start(start), end(end)

{}

Eigen::Vector2d Wall::get_start() const {
  return start;
}

Eigen::Vector2d Wall::get_end() const {
  return end;
}

Eigen::Vector2d Wall::dir_vec() const {
  return end - start;
}

std::vector<Wall> create_map(const std::vector<std::tuple<Eigen::Vector2d, Eigen::Vector2d>> walls) {
  std::vector<Wall> map_walls;
  map_walls.reserve(walls.size());
  /*for (size_t i = 0; i < walls.size(); i++) {
    Eigen::Vector2d wall_start = std::get<0>(walls[i]);
    Eigen::Vector2d wall_end = std::get<1>(walls[i]);
    map_walls.push_back(Wall(wall_start, wall_end));
  }*/
  for (const auto& [start, end] : walls){
    map_walls.emplace_back(start, end);
  }
  return map_walls;
}
}

File: python/map.py
import numpy as np

class Wall:
    def __init__(self, start, end):
        self.start = start
        self.end = end
    def dir_vec(self):
        return self.end - self.start

def create_map(walls):
    map_walls = []
    for wall in walls:
        map_walls.append(Wall(np.array([wall[0][0],wall[0][1]]), np.array([wall[1][0], wall[1][1]])))
        #print(map_walls[0].start)
    return map_walls
